File: nearly-spring-api/src/main/java/com/ukrainianboyz/nearly/model/entity/SecureUser.java
package com.ukrainianboyz.nearly.model.entity;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Value;

@AllArgsConstructor
@Value
@Getter
public class SecureUser {

    String userId;

    String userName;

    String userBio;

    String imageUrl;

    Integer statusIndicator;

    String appUserId;
}

File: nearly-android/app/src/main/java/com/ukrainianboyz/nearly/model/SecureUser.kt
package com.ukrainianboyz.nearly.model

import com.ukrainianboyz.nearly.entity.Friend
import com.ukrainianboyz.nearly.utils.apiUtils.FriendRequestStatus
import com.ukrainianboyz.nearly.utils.apiUtils.UserStatus

data class SecureUser (
    val userId: String,
    val userName: String,
    val userBio: String,
    val imageUrl: String?,
    val statusIndicator: Int,
    var appUserId: String
)


fun SecureUser.toFriend(requestStatus: FriendRequestStatus): Friend{
    return Friend(userId,userName,userBio,imageUrl, UserStatus.fromInt(statusIndicator),requestStatus, appUserId)
}